### PR TITLE
New animations 

### PR DESCRIPTION
**Pull request by lhupe**
OK, this is it! The cool new animation function which will provide real-time animations for several
particles with *all* the features (to be implemented)

Also fixed two bugs with #166 and changes the behaviour of `timeseries` to allow for animations
of pinned particles

## Still missing

- [x] Support for frame saving & ffmpeg
- [x] Custom figsizes & DPIs
- [x] Custom options for plotting the particle and tail (this might require a small change to `plot(p::AbstractParticle)` so I can use this function without using an actual `AbstractParticle`)
- [ ] new docs

### CODE REVIEW

**Comment by Datseris** [comment]
Can you paste example code I can just copy paste and marvel away?

**Comment by lhupe** [comment]
Here you go!
```julia
using PyPlot, DynamicalBilliards
bd, rs = billiard_raysplitting_showcase()
ps = [randominside(bd), randominside(bd, 2.0)]
animate_evolution(ps, bd, 10.0, rs)
```

**Comment by Datseris** [comment]
hooooly shit this is **insane**

File: src/plotting/animations.jl [hunk 1]
```diff
@@ -1,109 +1,98 @@
...
+  * `dt = 0.01` : Time resolution used for production of time series(see 
+    [`timeseries`](@ref). It is not recommended to increase this value to 
+    preserve the accuracy of the plots.
+  * `frameskip = 5` : The amount of `dt`-steps in between succesive frames
+  * `tailtime = 1.0` : The length of the "tail" trailing the particle in time 
+    units
+"""
+function animate_evolution(ps::AbstractVector{<:AbstractParticle{T}}, bd::Billiard,
+                           t, raysplitters = nothing; dt  = 0.01,
+                           frameskip = 5, tailtime = 1.0, ) where {T}
```
Comment (Datseris) [inline src/plotting/animations.jl]: I think it is good to keep the old behavior of `animate` with respect to the figure: if it wasn't given a specific `axis`, it was creating a new figure, getting current axis and then also calling `plot(bd)` on it.
```suggestion
                           frameskip = 5, tailtime = 1.0, 
                           ax = (figure(); plot(bd); gca())) where {T}
```

**Comment by Datseris** [comment]
Why is all particle orbits plotted with same color? I wonder if it is easy to restore the matplotlib color cycle functionality

**Comment by lhupe** [comment]
I didn't really about colors and axes while I wrote this, I'll add all of that over the next days. 
I think the normal color cycle could easily be restored, but maybe we should consider skipping `C2` (as it is the default boundary color [or is it `C3`?])

**Comment by Datseris** [comment]
None of the above: `obcolor(::Obstacle) = (0,0.6,0)`. I think it is fine without skipping anything; just make it as simple as possible

**Comment by lhupe** [comment]
Also, I would like to rewrite `plot(p: AbstractParticle)` to use a function `plot_particle(pos, vel)`. At the moment, I' have literally copy-pasted code from `plot_particle` into the `setup_animation` function

File: src/plotting/animations.jl [hunk 2]
```diff
@@ -1,109 +1,113 @@
...
+                           tail_kwargs = NamedTuple()) where {T}
+
+    nps = length(ps)
+    taillength = round(Int, tailtime/dt)
+
+    ax = PyPlot.gca()
+    plot(bd, ax = ax)
+    
+    plotframes = Vector{Function}(undef, nps)
+    skipframes = Vector{Function}(undef, nps)
```
Comment (Datseris) [inline src/plotting/animations.jl]: That is okay for now, but not a good long-term solution, as this object is highly unstable.

A better design is to save the line structs in arrays and define genereic functions that take xt,yt etc.

Comment (Datseris) [reply src/plotting/animations.jl]: But don't do anything about it, leave it as is for now since it "works".

**Comment by Datseris** [comment]
Amazing. Only thing left is the new logo. I'll do it next week as it requires using photoshop,..,.,..,

**Pull request merged**